docs/source/install_rocm.rst

```diff
@@ -0,0 +1,149 @@
+[Experimental] Installation Guide for ROCm environemt
+=====================================================
+
+.. contents:: :local:
+
+This is experimental feature. We recommend that only advanced users use this.
+
+Recommended Environments
+------------------------
+
+We recommend the following Linux distributions.
+
+* `Ubuntu <https://www.ubuntu.com/>`_ 16.04 / 18.04 LTS (64-bit)
+
+
+Requirements
+------------
+
+You need to have the following components to use CuPy.
+
+* AMD GPU
+* `ROCm <https://rocm.github.io/install.html>`_
+    * Supported Versions: ROCm 2.6+.
+* `Python <https://python.org/>`_
+* `NumPy <http://www.numpy.org/>`_
+
+And please install ROCm libraries.
+
+::
+
+  $ sudo apt install hipblas hipsparse rocrand
+
+
+Before installing CuPy, we recommend you to upgrade ``setuptools`` and ``pip``::
+
+  $ pip install -U setuptools pip
+
+
+Install CuPy from Source
+------------------------
+
+It is recommended to use wheels whenever possible.
+However, if wheels cannot meet your requirements (e.g., you are running non-Linux environment or want to use a version of CUDA / cuDNN / NCCL not supported by wheels), you can also build CuPy from source.
+
+When installing from source, C++ compiler such as ``g++`` is required.
+You need to install it before installing CuPy.
+This is typical installation method for each platform::
+
+  # Ubuntu 16.04
+  $ apt-get install g++
+
+.. note::
+
+   If you upgrade or downgrade the version of ROCm, you may need to reinstall CuPy.
+   See :ref:`rocm_install_reinstall` for details.
+
+Using pip
+~~~~~~~~~
+
+You can install `CuPy package <https://pypi.python.org/pypi/cupy>`_ via ``pip``.
+
+::
+
+  $ export HCC_AMDGPU_TARGET=gfx900
+  $ export __HIP_PLATFORM_HCC__
+  $ export CUPY_INSTALL_USE_HIP=1
+  $ pip install cupy
+
+Using Tarball
+~~~~~~~~~~~~~
+
+The tarball of the source tree is available via ``pip download cupy`` or from `the release notes page <https://github.com/cupy/cupy/releases>`_.
+You can install CuPy from the tarball::
+
+  $ pip install cupy-x.x.x.tar.gz
+
+You can also install the development version of CuPy from a cloned Git repository::
+
+  $ git clone https://github.com/cupy/cupy.git
+  $ cd cupy
+  $ export HCC_AMDGPU_TARGET=gfx900
```

Comment:
The value of `HCC_AMDGPU_TARGET`, in this case, `gfx900` will be different for different AMD GPUs, right? If so, I think it's better to mention users have to change it according to the GPU.
```suggestion
  $ export HCC_AMDGPU_TARGET=gfx900  # This value should be changed based on your GPU
```